src/adam_core/dynamics/moid.py

```diff
@@ -21,14 +21,34 @@
 search for the overall minimum.
 """
 
+from typing import List, Union
+
 import numpy as np
 import numpy.typing as npt
+import quivr as qv
+import ray
+from ray import ObjectRef
 from scipy.optimize import minimize_scalar
 
-from ..coordinates import CartesianCoordinates, KeplerianCoordinates
+from ..coordinates import (
+    CartesianCoordinates,
+    KeplerianCoordinates,
+    Origin,
+    OriginCodes,
+)
 from ..dynamics.propagation import _propagate_2body
 from ..orbits import Orbits
+from ..propagator.utils import _iterate_chunks
```

Comment:
We should move this to some kind of iter utils module

Reply:
Done.